docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+# Specs for the ML.NET CLI with AutoML
+
+## Context
+
+**Commitments: This specs document is 100% aspirational and will change while it's being discussed and implementation is evolving based on feedback. There are no commitments derived from this document except for the first upcoming minor version at any given time (v0.1, initially).**
+
+The CLI will be branded as the ML.NET CLI since this CLI will also have additional features in addition to AutoML features.
+
+The .NET AutoML API (.NET based) will be part of the [ML.NET](https://github.com/dotnet/machinelearning) API.
+AutoML features will be used for certain important foundational features of the ML.NET CLI.
+
+This specs-doc focuses most of all on the CLI features related to AutoML, but it will also consider (in less detail) the scenarios where AutoML is not needed, so the CLI syntax will be consistent end-to-end for all the possible scenarios in the future. 
+
+# Problem to solve
+
+Customers (.NET developers) have tolds us through many channels that they can get started with [ML.NET](https://github.com/dotnet/machinelearning) and follow the initial simple examples. However, as soon as they have to create their own model to solve their problems, they are blocked because they don't know what learner/algorithms are better for them to pick and use, what hyper-parameters to use or even what data transformations they need to do.
+
+The subset of .NET developers who at the same time are skilled on data science and machine learning in general is very small compared to the total number of regular .NET developers.
```

Comment:
>compared to the total number of regular .NET developers. [](start = 125, length = 56)

"compared to the overall number of  .NET Developers."  -- "regular developer" is undefined term :)

Reply:
Right. I mean a "regular developer" at the time of writing this spec.
At the time when the "regular .NET developer" evolves (hopefully with much more skills on ML) this specs-doc will be stale and old. 

Note that this doc is a specs-document focusing on current goals. Such comments and goals won't be part of the official documentation for the CLI, logically.

Reply:
suggest just drop tehe "regular" and just use "compared to the overall number of .NET developers"

---
In reply to: [259508845](https://github.com/dotnet/machinelearning/pull/2693#discussion_r259508845) [](ancestors = 259508845)